File: Exercise1.py
import threading
import time


i = 0

def someThreadFunction1():
    global i
    for x in range (0,1000000):
        i = i + 1

def someThreadFunction2():
    global i
    for x in range (0, 1000000):
        i = i - 1


def main():
    global i
    someThread1 = threading.Thread(target = someThreadFunction1)
    someThread2 = threading.Thread(target = someThreadFunction2)

    someThread1.start()
    someThread2.start()

    someThread1.join()
    someThread2.join()
    
    print(i)


main()

File: 1/Exercise1.c


#include <pthread.h>
#include <stdio.h>

int i = 0;

// Note the return type: void*
void* someThreadFunction1(){
     int x;
     for (x = 0; x < 1000000; x++){
	i++;	
	}
    return NULL;
}

void* someThreadFunction2(){
    int x;
    for (x  = 0; x < 1000000; x++){
	i--;	
	}
    return NULL;
}


int main(){
    pthread_t someThread1;
    pthread_t someThread2;

    pthread_create(&someThread1, NULL, someThreadFunction1, NULL);
    pthread_create(&someThread2, NULL, someThreadFunction2, NULL);

    // Arguments to a thread would be passed here ---------^
    
    pthread_join(someThread1, NULL);
    pthread_join(someThread2, NULL);

    printf("%i\n", i);
    return 0;
    
}
